qiskit/algorithms/phase_estimators/ipe.py

```diff
@@ -117,7 +126,20 @@ def _estimate_phase_iteratively(self, unitary, state_preparation):
         # k runs from the number of iterations back to 1
         for k in range(self._num_iterations, 0, -1):
             omega_coef /= 2
-            if self._quantum_instance.is_statevector:
+
+            if self._sampler is not None:
+
+                qc = self.construct_circuit(
+                    unitary, state_preparation, k, -2 * numpy.pi * omega_coef, True
+                )
+                try:
+                    sampler_job = self._sampler.run([qc], shots=self.shots)
+                    result = sampler_job.result().quasi_dists[0]
```

Comment:
Do we need to take any additional care anywhere in these algos given that sampler result is a quasi distribution with possible negative values? I'm thinking no but just want to double check - @jlapeyre

Reply:
The result is only used in this line which compares 2 sampled values; it should work ok.
`x = 1 if result.get(1, 0) > result.get(0, 0) else 0`